core/src/main/scala/kafka/log/Log.scala

```diff
@@ -461,6 +461,20 @@ class Log(@volatile var dir: File,
    * </ol>
    */
   private def analyzeAndValidateRecords(records: MemoryRecords): LogAppendInfo = {
+    def validateEntry(entry: LogEntry) {
```

Comment:
What's the motivation for this change? This method seems to be called only once, and I am not sure the indirection is worth it.